certificate-service/app/src/main/java/se/inera/intyg/certificateservice/application/citizen/service/converter/CertificateTextConverter.java

```diff
@@ -0,0 +1,36 @@
+package se.inera.intyg.certificateservice.application.citizen.service.converter;
+
+import java.util.Objects;
+import org.springframework.stereotype.Component;
+import se.inera.intyg.certificateservice.application.citizen.dto.CertificateLinkDTO;
+import se.inera.intyg.certificateservice.application.citizen.dto.CertificateTextDTO;
+import se.inera.intyg.certificateservice.application.citizen.dto.CertificateTextTypeDTO;
+import se.inera.intyg.certificateservice.domain.common.model.CertificateText;
+import se.inera.intyg.certificateservice.domain.common.model.CertificateTextType;
+
+@Component()
+public class CertificateTextConverter {
+
+  public CertificateTextDTO convert(CertificateText certificateText) {
+    return CertificateTextDTO.builder()
+        .text(certificateText.text())
+        .type(convertCertificateTextType(certificateText.type()))
+        .links(certificateText.links().stream()
+            .map(link -> CertificateLinkDTO.builder()
+                .url(link.url())
+                .id(link.id())
+                .name(link.name())
+                .build()
+            )
+            .toList())
+        .build();
+  }
+
+  private CertificateTextTypeDTO convertCertificateTextType(
+      CertificateTextType certificateTextType) {
+    if (Objects.requireNonNull(certificateTextType) == CertificateTextType.PREAMBLE_TEXT) {
+      return CertificateTextTypeDTO.PREAMBLE_TEXT;
+    }
+    return null;
```

Comment:
Don't we need to have a texttype? Should we instead throw exception here to fail fast, instead of an error happening on the client side?